glass/shells.py

```diff
@@ -235,10 +235,10 @@ def tophat_windows(
     ws = []
     for zmin, zmax in itertools.pairwise(zbins):
         n = max(round((zmax - zmin) / dz), 2)
-        z = np.linspace(zmin, zmax, n)
-        w = wht(z)  # type: ignore[arg-type]
+        z = np.linspace(zmin, zmax, n, dtype=np.float64)
+        w = wht(z)
         zeff = np.trapezoid(w * z, z) / np.trapezoid(w, z)
```

Comment:
`np.trapezoid` can return an array and I don't think they have enough overloads in place -- therefore `zeff` can be an array and the type signature of `RadialWindow` should reflect this